src/ReturnTypes/AuthExtension.php

```diff
@@ -59,7 +59,7 @@ public function getTypeFromStaticMethodCall(
             ->get('config');
 
         if ($userModel = $config->get('auth.providers.users.model')) {
-            return new ObjectType($userModel);
+            return TypeCombinator::addNull(new ObjectType($userModel));
```

Comment:
When the user is not logged, the `User::auth()` returns `null`?

Reply:
Correct, here is an example in the default `SessionGuard` that laravel provides: https://github.com/laravel/framework/blob/5.7/src/Illuminate/Auth/SessionGuard.php#L115-L117